smartsim/settings/pbsSettings.py

```diff
@@ -185,40 +213,71 @@ def format_batch_args(self) -> t.List[str]:
             opts += [" ".join((prefix + opt, str(value)))]
         return opts
 
+    def _sanity_check_resources(
+            self,
+            resources: t.Optional[t.Dict[str, t.Union[str,int]]] = None
+    ) -> None:
+        """Check that only select or nodes was specified in resources
+
+        Note: For PBS Pro, nodes is equivalent to 'select' and 'place' so
+        they are not quite synonyms. Here we assume that
+        """
+        checked_resources = resources if resources else self.resources
+
+        has_select = checked_resources.get("select", None)
+        has_nodes = checked_resources.get("nodes", None)
+
+        if has_select and has_nodes:
+            raise SSConfigError(
+                "'select' and 'nodes' cannot both be specified. This can happen "
+                "if nodes were specified using the 'set_nodes' method and "
+                "'select' was set using 'set_resource'. Please only specify one."
+            )
+
+        if has_select and not isinstance(has_select, int):
+            raise TypeError("The value for 'select' must be an integer")
+        if has_nodes and not isinstance(has_nodes, int):
+            raise TypeError("The value for 'nodes' must be an integer")
+
+        for key, value in checked_resources.items():
+            allowed_types = [int, str]
+            if not any(isinstance(key, type) for type in allowed_types):
+                raise TypeError(
+                    f"The type of {key=} is {type(key)}. Only int and str "
+                    "are allowed."
+                )
+            if not any(isinstance(value, type) for type in allowed_types):
+                raise TypeError(
+                    f"The value associated with {key=} is {type(value)}. Only int "
+                    "and str are allowed."
+                )
```

Comment:
No need to loop over the valid types: `isinstance` can take a tuple:
```suggestion
            if not isinstance(key, str):
                raise TypeError(
                    f"The type of {key=} is {type(key)}. Only int and str "
                    "are allowed."
                )
            if not isinstance(value, (int, str)):
                raise TypeError(
                    f"The value associated with {key=} is {type(value)}. Only int "
                    "and str are allowed."
                )
```